### DISCUSSION TITLE
alt-i/h toggles in fzf.lua doesn't work on MacOS

### DISCUSSION THREAD

**Discussion by Jomik** [LazyVim/LazyVim#4029] [Q&A] [answered]
Hey,
I just switched to fzf.lua, and saw that I can use `alt+h` to toggle hidden files, which I definitely need to search into `.github/` and such.
But `alt-h` on my MacOS simply outputs `˙`, which is not what we need.

Do anyone have any solution on how to make this work on MacOS? Changing to ctrl-h does not seem to work either, as that is resolved as a backspace..

Thanks!

**Comment by dpetka2001** [comment]
Putting something like this in your fzf-lua spec should work
```lua
      local actions = require("fzf-lua.actions")
      opts.files = {
        actions = {
          ["ctrl-i"] = { actions.toggle_ignore },
          ["ctrl-h"] = { actions.toggle_hidden },
        },
      }
```
Change `ctrl-h` to whatever keymap you can alternatively use on Mac. I don't have any experience with Mac, so I don't know how keys are interpreted there. You should experiment to see what key sequence will work for you. The example I provided is more as to how you can change the keys in LazyVim to whatever you want. Have you tried `<M-h>` instead of `<alt-h>` to see if this maybe works?

**Reply by Jomik** [reply]
I am thinking I may have to use `M` instead. But I cannot get it to work in the fzf-lua spec.
I am trying to do...
```lua
  {
    "ibhagwan/fzf-lua",
    optional = true,
    opts = function(_, opts)
      local actions = require("fzf-lua.actions")
      return vim.tbl_deep_extend("force", opts, {
        files = {
          cwd_prompt = false,
          actions = {
            ["M-i"] = { actions.toggle_ignore },
            ["M-h"] = { actions.toggle_hidden },
          },
        },
        grep = {
          actions = {
            ["M-i"] = { actions.toggle_ignore },
            ["M-h"] = { actions.toggle_hidden },
          },
        },
      })
    end,
  },
```

**Reply by dpetka2001** [reply]
No idea really. It may have to do with what modifiers `fzf` can accept. Not sure. Just try to find a key that works. Or you could try using `kitty` or `wezterm` where I didn't have any problems with the `Alt` modifier. It seems to be an `iterm` specific thing and I have no idea how you can get around it, since I don't have Mac.

**Reply by Jomik** [reply]
Thank you. :)

**Comment by quangdn42** [comment, marked as answer]
I'm pretty sure the issue is how your terminal emulator sees your hotkeys. This might give you a clue https://wezfurlong.org/wezterm/config/key-encoding.html#kitty-keyboard-protocol

**Reply by Jomik** [reply]
This is essentially the answer. I use Neovide, and realise that there is an option I need to set
https://neovide.dev/configuration.html?highlight=alt#macos-option-key-is-meta
`vim.g.neovide_input_macos_option_key_is_meta = 'only_left'`
There is a similar option in iterm and most other terminal emulators.

**Reply by alanramsay** [reply]
Thanks for this.  For users of ghostty, the answer is https://ghostty.org/docs/config/reference#macos-option-as-alt

**Reply by oathlesss** [reply]
If you are using aerospace check the `aerospace.toml` for conflicting keybinds.
